### PR TITLE
aws - bedrock customization jobs, tag, kms-key and stop action

### PR DESCRIPTION
**Pull request by PratMis**
This PR adds support for aws bedrock customization jobs and would help enforce tagging, vpcConfig and encryption on them. I have made sure we only look at jobs that are `InProgress` else we would end up capturing too much noise imo. There's also a stop action associated with the resources.

Also added a `kms-key` filter for bedrock custom models

Todo:

- [x] tests

### CODE REVIEW

File: tests/data/placebo/test_bedrock_agent_encryption/bedrock-agent.GetAgent_1.json [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                "__class__": "datetime",
+                "year": 2024,
+                "month": 2,
+                "day": 15,
+                "hour": 11,
+                "minute": 2,
+                "second": 55,
+                "microsecond": 216611
+            },
+            "promptOverrideConfiguration": {
```
Comment (PratMis) [inline tests/data/placebo/test_bedrock_agent_encryption/bedrock-agent.GetAgent_1.json:46]: We could potentially drop `promptOverrideConfiguration`.

Comment (PratMis) [reply tests/data/placebo/test_bedrock_agent_encryption/bedrock-agent.GetAgent_1.json:46]: Dropped it from getting captured in the resources file

File: c7n/resources/bedrock.py [hunk 2]
```diff
@@ -121,3 +122,208 @@ def process(self, resources):
...
+                - type: stop
+
+    """
+    schema = type_schema('stop')
+    permissions = ('bedrock:StopModelCustomizationJob',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('bedrock')
+        for r in resources:
+            client.stop_model_customization_job(jobIdentifier=r['jobArn'])
```
Comment (kapilt) [inline c7n/resources/bedrock.py:281]: this probably needs a catch around state, ie does stopping a stopped job result in error?

File: c7n/resources/resource_map.py [hunk 3]
```diff
@@ -165,6 +166,7 @@
   "aws.message-broker": "c7n.resources.mq.MessageBroker",
   "aws.message-config": "c7n.resources.mq.MessageConfig",
   "aws.mirror-session": "c7n.resources.vpc.TrafficMirrorSession",
+  "aws.model-customization-job": "c7n.resources.bedrock.BedrockModelCustomizationJob",
```
Comment (kapilt) [inline c7n/resources/resource_map.py:169]: hmm.. would like to keep this prefixed with bedrock, there's a few model / job services in aws already. maybe `bedrock-customization-job` ?

File: c7n/resources/bedrock.py (hunk 2 above)
Comment (PratMis) [reply c7n/resources/bedrock.py:281]: Thanks for the feedback @kapilt . I did try to run the stop operation in loop and on already stopped jobs. Apparently, it doesn't trigger any exceptions. 
```
>>> client = boto3.client('bedrock', 'us-east-1')
>>> response = client.stop_model_customization_job(jobIdentifier='arn:aws:bedrock:us-east-1:644160558196:model-customization-job/cohere.command-text-v14:7:4k/kktbyp9iwhsw')
>>> response = client.stop_model_customization_job(jobIdentifier='arn:aws:bedrock:us-east-1:644160558196:model-customization-job/cohere.command-text-v14:7:4k/kktbyp9iwhsw')
>>> response = client.stop_model_customization_job(jobIdentifier='arn:aws:bedrock:us-east-1:644160558196:model-customization-job/cohere.command-text-v14:7:4k/kktbyp9iwhsw')
```
Interestingly, I also read the doc re: [exceptions](https://docs.aws.amazon.com/bedrock/latest/APIReference/API_StopModelCustomizationJob.html), and found a `ConflictException` but wasn't able to create a scenario that would trigger it

File: c7n/resources/resource_map.py (hunk 3 above)
Comment (PratMis) [reply c7n/resources/resource_map.py]: Makes sense to me. Addressed

**Pull request merged**